File: platform/android-seller/app/src/main/java/ca/shawcliffe/seller/PreviewModels.kt
package ca.shawcliffe.seller

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Read-only shapes for the "Preview" tab, which shows the seller what their
 * customers see. Mirrors ios-customer's storefront models; kept separate from
 * the seller's own [DailyStatus] (which only carries the fields the status
 * toggle writes) since this needs the fuller display-oriented column set.
 */
@Serializable
data class ClientBranding(
    @SerialName("client_id") val clientId: String,
    @SerialName("primary_color") val primaryColor: String? = null,
    @SerialName("logo_url") val logoUrl: String? = null,
    @SerialName("app_name") val appName: String? = null,
    val tagline: String? = null,
)

@Serializable
data class StorefrontLocation(
    @SerialName("client_id") val clientId: String,
    val id: String,
    @SerialName("display_name") val displayName: String,
    val address: String? = null,
    @SerialName("parking_notes") val parkingNotes: String? = null,
)

@Serializable
data class StorefrontStatus(
    @SerialName("client_id") val clientId: String,
    val id: String,
    val date: String,
    val status: String,
    @SerialName("hours_open") val hoursOpen: String? = null,
    @SerialName("hours_close") val hoursClose: String? = null,
    @SerialName("location_id") val locationId: String? = null,
    @SerialName("custom_message") val customMessage: String? = null,
)

File: clients/little-dough-bread-co/ios-seller/ShawcliffeSeller/PreviewModels.swift
import Foundation

/// Read-only shapes for the "Preview" tab, which shows the seller what their
/// customers see. Mirrors ios-customer's storefront models; kept separate
/// from the seller's own `DailyStatus` (which only carries the fields the
/// status toggle writes) since this needs the fuller display-oriented column set.
struct ClientBranding: Codable {
    let clientId: String
    var primaryColor: String?
    var logoUrl: String?
    var appName: String?
    var tagline: String?

    enum CodingKeys: String, CodingKey {
        case clientId = "client_id"
        case primaryColor = "primary_color"
        case logoUrl = "logo_url"
        case appName = "app_name"
        case tagline
    }
}

struct StorefrontLocation: Codable, Identifiable {
    let clientId: String
    let id: String
    let displayName: String
    let address: String?
    let parkingNotes: String?

    enum CodingKeys: String, CodingKey {
        case clientId = "client_id"
        case id
        case displayName = "display_name"
        case address
        case parkingNotes = "parking_notes"
    }
}

struct StorefrontStatus: Codable {
    let clientId: String
    let id: String
    let date: String
    var status: DailyStatusValue
    var hoursOpen: String?
    var hoursClose: String?
    var locationId: String?
    var customMessage: String?

    enum CodingKeys: String, CodingKey {
        case clientId = "client_id"
        case id
        case date
        case status
        case hoursOpen = "hours_open"
        case hoursClose = "hours_close"
        case locationId = "location_id"
        case customMessage = "custom_message"
    }
}
